akd/src/storage/manager/mod.rs

```diff
@@ -421,17 +420,16 @@ impl<Db: Database> StorageManager<Db> {
         }
     }
 
-    /// Tombstone a set of records adhereing to the caching + transactional
-    /// settings of the storage manager
-    pub async fn tombstone_value_states(&self, keys: &[ValueStateKey]) -> Result<(), StorageError> {
-        if keys.is_empty() {
-            return Ok(());
-        }
-
-        let data = self.batch_get::<ValueState>(keys).await?;
+    /// Tombstones all value states for a given AkdLabel, up to and including a given epoch
+    pub async fn tombstone_value_states(
+        &self,
+        username: &AkdLabel,
+        epoch: u64,
+    ) -> Result<(), StorageError> {
+        let key_data = self.get_user_data(username).await?;
         let mut new_data = vec![];
```

Comment:
Nit: I think you can effectively write this a bit cleaner (without a mutable variable) by doing something like so:

```rust
    let new_data = key_data
        .states
        .into_iter()
        .filter_map(|vs| {
            if vs.epoch <= epoch && vs.value.0 != crate::TOMBSTONE {
                DbRecord::ValueState(ValueState {
                    epoch: value_state.epoch,
                    label: value_state.label,
                    value: crate::AkdValue(crate::TOMBSTONE.to_vec()),
                    username: value_state.username,
                    version: value_state.version,
                })
            } else {
                None
            }
        })
        .collect();
```

Reply:
Ah, the rest of this file uses for loops and mutable collections, I'll keep it the current way for consistency for now. Thanks for the suggestion!